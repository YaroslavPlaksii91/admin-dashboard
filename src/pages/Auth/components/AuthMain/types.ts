import { ReactNode } from 'react';

export type AuthMainProps = {
  children: ReactNode;
};
