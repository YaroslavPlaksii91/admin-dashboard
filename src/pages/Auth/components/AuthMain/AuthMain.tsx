import { FC } from 'react';

import { AuthMainProps } from './types';

import styles from './AuthMain.module.css';

export const AuthMain: FC<AuthMainProps> = ({ children }) => {
  return <main className={styles.main}>{children}</main>;
};
