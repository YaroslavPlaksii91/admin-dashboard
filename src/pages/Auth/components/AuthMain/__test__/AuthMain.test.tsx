import { cleanup, render, screen } from '@testing-library/react';

import { AuthMain } from '../AuthMain';

describe('AuthMain component', () => {
  afterEach(cleanup);

  const text = 'Content';
  const Content = () => <div>{text}</div>;

  it('should be rendered', () => {
    const { asFragment } = render(
      <AuthMain>
        <Content />
      </AuthMain>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders children', () => {
    render(
      <AuthMain>
        <Content />
      </AuthMain>,
    );

    expect(screen.getByText(text)).toBeInTheDocument();
  });
});
