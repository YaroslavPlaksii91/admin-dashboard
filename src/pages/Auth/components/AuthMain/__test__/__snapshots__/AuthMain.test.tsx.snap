// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthMain component should be rendered 1`] = `
<DocumentFragment>
  <main>
    <div>
      Content
    </div>
  </main>
</DocumentFragment>
`;
